DMCompiler/DM/Expressions/Builtins.cs

```diff
@@ -82,6 +82,8 @@ sealed class NewPath : DMExpression {
             proc.PushType(typeId);
             proc.CreateObject(argumentInfo.Type, argumentInfo.StackSize);
         }
+
+        public override DreamPath? Path => _targetPath;
```

Comment:
This code snippet compiles fine in BYOND, showing that this is actually treated as `(new /datum/test2):foo`
```
/datum/test/var/foo = "foo"
/datum/test2/var/bar = "bar"

/proc/main()
    world.log << ((new /datum/test2).foo)
```

You could achieve this by adding `case NewPath:` [here](https://github.com/OpenDreamProject/OpenDream/blob/master/DMCompiler/DM/Visitors/DMVisitorExpression.cs#L476). Though I wonder if there'd be no harm in the additional type safety this solution provides?

Reply:
This is fine as-is, the additional type safety is better and nothing depends on this to my knowledge.